internal/iam/command_ipfilter_update.go

```diff
@@ -91,52 +97,79 @@ func (c *ipFilterCommand) update(cmd *cobra.Command, args []string) error {
 		updateIpFilter.ResourceGroup = &resourceGroup
 	}
 
-	// For each IP group ID being added that isn't in the existing slice, append it to the new slice
-	if len(addIpGroups) > 0 {
-		for _, ipGroupId := range addIpGroups {
-			if !slices.Contains(newIpGroupIds, ipGroupId) {
-				newIpGroupIds = append(newIpGroupIds, ipGroupId)
-			}
+	// Create a map of the current IP group IDs
+	currentIpGroupIdsMap := make(map[string]Operation)
+	for _, ipGroupId := range currentIpGroupIds {
+		currentIpGroupIdsMap[ipGroupId] = ADD
+	}
+
+	// Create a map of the new IP group ID values
+	newIpGroupIdsMap := make(map[string]Operation)
+	// Add each new IP Group ID to the map
+	for _, ipGroupId := range addIpGroups {
+		newIpGroupIdsMap[ipGroupId] = ADD
+	}
+
+	// For each IP group ID being removed that is in the new map, set its key to REMOVE
+	for _, ipGroupId := range removeIpGroups {
+		if newIpGroupIdsMap[ipGroupId] == ADD {
+			log.CliLogger.Warn("Attempting to add and remove the same IP group.")
 		}
+		newIpGroupIdsMap[ipGroupId] = REMOVE
```

Comment:
Not sure we need the enums here. Can't we just `delete()` the entry from the map if it's in the `removeIpGroups` array?

Reply:
I believe we need the enums to differieniate between a value we are removing versus a value that doesn't exist in the map.

```		if (newCidrBlocksMap[cidrBlock] == REMOVE) && (value == NONE) {
			log.CliLogger.Warn("Attempting to remove a CIDR block that does not exist on this IP group.")
		}
		```
That allows us to log the warning if they try to remove an IP group that doesn't exist in the original map.
Without this warning, then yes, we can just delete everything instead of having to log it as "REMOVE"

Reply:
How about we have three maps, one for existing values, one for values to add, and one for values to delete? For the first warning, we check if any of the values in the "to delete" map is also present in the "to add" map. For the second warning, we check if any of the values in the "to delete" map is not present in the existing values map.

Reply:
Sure, I can get right on this.